### PR TITLE
Add Stackdriver Trace sample code for the doc

### PR DESCRIPTION
**Pull request by tmatsuo**
cc @chingor13

### CODE REVIEW

File: testing/run_test_suite.sh [hunk 1]
```diff
@@ -65,6 +65,7 @@ do
         fi
     fi
     pushd ${DIR}
+    mkdir -p build/logs
```
Comment (bshaffer) [inline testing/run_test_suite.sh]: 🎉

File: trace/trace-sample.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# [END trace_use_statement]
+
+$projectId = getenv('GOOGLE_CLOUD_PROJECT');
+if ($projectId === false) {
+    die('Set GOOGLE_CLOUD_PROJECT envvar');
+}
+
+# [START exporter_setup]
+$exporter = new StackdriverExporter(
```
Comment (bshaffer) [inline trace/trace-sample.php]: It's not in the PSR coding standards, but I've been doing these like this, so it'd be better for internal sample consistency:

```php
$exporter = new StackdriverExporter([
    'clientConfig' => [
        'projectId' => $projectId
    ]
]);
```

File: trace/trace-sample.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+if (getenv('USE_NULL_EXPORTER')) {
+    $exporter = new NullExporter();
+}
+# [START tracer_start]
+Tracer::start($exporter);
+# [END tracer_start]
+
+function use_trace()
+{
+    # [START span_with_closure]
```
Comment (bshaffer) [inline trace/trace-sample.php]: The reference block here is off.. the `END` ref should be inside the function

File: trace/trace-sample.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    die('Set GOOGLE_CLOUD_PROJECT envvar');
+}
+
+# [START exporter_setup]
+$exporter = new StackdriverExporter(
+    ['clientConfig' => [
+        'projectId' => $projectId
+    ]]);
+# [END exporter_setup]
+// For testing
```
Comment (bshaffer) [inline trace/trace-sample.php]: How about this comment instead? 

```php
// When running tests, use a null exporter instead.
```

But a better question is, why not just use `StackdriverExporter`?

Or, in the tests, just call `Trace::start(new NullExporter);` before calling `use_trace()`?

File: trace/trace-sample.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+if (getenv('USE_NULL_EXPORTER')) {
+    $exporter = new NullExporter();
+}
+# [START tracer_start]
+Tracer::start($exporter);
+# [END tracer_start]
+
+function use_trace()
```
Comment (bshaffer) [inline trace/trace-sample.php]: I would like a more descriptive name for this function here. How about `trace_callable()`?

File: trace/trace-sample.php (hunk 2 above)
Comment (tmatsuo) [reply trace/trace-sample.php]: Done (with the next commit)

File: trace/trace-sample.php (hunk 3 above)
Comment (tmatsuo) [reply trace/trace-sample.php]: done

File: trace/trace-sample.php (hunk 4 above)
Comment (tmatsuo) [reply trace/trace-sample.php]: Having integration test is good in general, but unfortunately now there's no way to fetch the trace from the API. I would just have the unit test like this for now.

File: trace/trace-sample.php (hunk 5 above)
Comment (tmatsuo) [reply trace/trace-sample.php]: Done

**Comment by tmatsuo** [comment]
@bshaffer Thanks, PTAL

**Comment by chingor13** [comment]
LGTM

**Pull request merged**